From: Chad Dupuis <cdupuis@redhat.com>
Date: Wed, 17 Nov 2010 20:56:13 -0500
Subject: [scsi] qla2xxx: check null fcport in _queuecommands
Message-id: <20101117205613.8117.82010.sendpatchset@localhost.localdomain>
Patchwork-id: 29473
O-Subject: [RHEL 5.6 PATCH v2] qla2xxx: Add check for null fcport in
	queuecommand functions.
Bugzilla: 644863
RH-Acked-by: Mike Christie <mchristi@redhat.com>

This is version 2 of the original patch:
qla2xxx: Add check for null fcport in qla24xx_queuecommand().

This patch adds the check for the null fcport to the qla2x00_queuecommand()
function in addition to the qla24xx_queuecommand() function.

Brew build for this patch was 2899688.

>From 92f6541e099818736131c34a5d9affa528adaafc Mon Sep 17 00:00:00 2001
From: root <root@lotus.lab.bos.redhat.com>
Date: Wed, 17 Nov 2010 09:57:49 -0500
Subject: [PATCH] qla2xxx: Add check for null fcport in queuecommand functions.


diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index bcf97cd..a3cebff 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -475,6 +475,11 @@ qla2x00_queuecommand(struct scsi_cmnd *cmd, void (*done)(struct scsi_cmnd *))
 	}
 
 	/* Close window on fcport/rport state-transitioning. */
+	if (!fcport) {
+		cmd->result = DID_NO_CONNECT << 16;
+		goto qc_fail_command;
+	}
+
 	if (fcport->drport) {
 		cmd->result = DID_IMM_RETRY << 16;
 		goto qc_fail_command;
@@ -546,6 +551,11 @@ qla24xx_queuecommand(struct scsi_cmnd *cmd, void (*done)(struct scsi_cmnd *))
 	}
 
 	/* Close window on fcport/rport state-transitioning. */
+	if (!fcport) {
+		cmd->result = DID_NO_CONNECT << 16;
+		goto qc24_fail_command;
+	}
+
 	if (fcport->drport) {
 		cmd->result = DID_IMM_RETRY << 16;
 		goto qc24_fail_command;
